forgot_password/handlers.py

```diff
@@ -22,10 +27,66 @@
 from .util import email as emailutil
 from .util import user as userutil
 
-
 logger = logging.getLogger(__name__)
 
 
+class IllegalArgumentError(ValueError):
+    pass
+
+
+ResetPasswordRequestParams = namedtuple(
+    'ResetPasswordRequestParams',
+    ['code', 'user_id', 'expire_at', 'user', 'user_record'])
+
+
+def download_template(url, name):
+    path_prefix = 'templates/forgot_password'
+    path = '/'.join([path_prefix, name])
+
+    Path(path_prefix).mkdir(parents=True, exist_ok=True)
```

Comment:
Now I come to think of it... do you think it is better to load the template on demand rather than loading it at launch? We could have a jinja loader that fetches and caches a template when sending the email.

This also have the added benefit that if the download failed the first time, we can retry when the plugin tries to send another email.